Quantum-Control-Applications-QuAM/Superconducting/quam_libs/experiments/two_qubit_xeb/README.md

```diff
@@ -0,0 +1,205 @@
+
+## How does the script work?
+
+The protocol relies on the OPX's ability to generate and execute in real-time random circuits on the quantum computer and compare the results with the ideal simulation. 
+The advantage of using QUA is that the randomization of the circuits can be done in parallel to circuit execution through real-time processing and random sampling.
+
+The user can choose which gate set to use to generate random unitaries. Usually, the experiment is performed with layers of random single-qubit gates followed by one fixed two-qubit gate, as depicted in the circuit below with the gate set formed by the gates $SX$, $SY$, and $SW$ [2]:
+[![XEB Circuit](images/xeb_circuit.png)](images/xeb_circuit.png)
+
+In this circuit, we have used the $SW$ gate defined in [2] as: 
+$$
+W^{1 / 2} \equiv R_{X+Y}(\pi / 2)=\frac{1}{\sqrt{2}}\left[\begin{array}{cc}
+1 & -\sqrt{i} \\
+\sqrt{-i} & 1
+\end{array}\right]
+$$
+The script will generate random circuits with the chosen gate set and run them on the quantum computer. The script will then calculate the cross-entropy between the ideal and actual probability distributions to estimate the layer fidelity (from which the two-qubit gate fidelity can be inferred).
+
+As opposed to Randomized Benchmarking, we do not invert the randomly generated circuit by applying an inverse, but we rather perform a fidelity estimation over the statistics of the outcomes when measuring the system in the computational basis.
+
+There are therefore four steps in the script:
+- Random circuits generation: Done within QUA in real-time, the script generates random circuits of different depths with the chosen gate set. At the same time, the gates sampled in real-time are streamed back to the classical side for the theoretical simulation. The circuits are randomized through the choice of the single qubit gates that are applied to each qubit at each layer (depth).
+- Execution: For each random sequence of gates of varying depths, the script runs the circuit on the quantum computer while leveraging real-time pulse modulation of the OPX. This is particularly useful for playing all possible random gates through one single gate baseline (usually the $SX$ ($X/2)$ gate). Alternatively, it is possible to play dedicated pulses for each of the random gates, through the use of a QUA `switch`statement (at the expense of possible more latency, depending on your gate durations). 
+- Theoretical simulation: The script simulates the random circuits on the classical side to compute the theoretical quantum state, from which we deduce the probabilities of getting all possible measurement outcomes.
+- Cross-entropy calculation: The script computes the cross-entropy between the expected and empirically deduced probability distributions to estimate the layer fidelity.
```

Comment:
Could you enumerate these? i.e., 

1. **Random circuits generation**: Done within QUA in real-time, the script generates random circuits of different depths with the chosen gate set. At the same time, the gates sampled in real-time are streamed back to the classical side for the theoretical simulation. The circuits are randomized through the choice of the single qubit gates that are applied to each qubit at each layer (depth).
2. **Execution:** For each random sequence of gates of varying depths, the script runs the circuit on the quantum computer while leveraging real-time pulse modulation of the OPX. This is particularly useful for playing all possible random gates through one single gate baseline (usually the $SX$ ($X/2)$ gate). Alternatively, it is possible to play dedicated pulses for each of the random gates, through the use of a QUA `switch`statement (at the expense of possible more latency, depending on your gate durations). 
3. **Theoretical simulation:** The script simulates the random circuits on the classical side to compute the theoretical quantum state, from which we deduce the probabilities of getting all possible measurement outcomes.
4. **Cross-entropy calculation:** The script computes the cross-entropy between the expected and empirically deduced probability distributions to estimate the layer fidelity.